isyfact-standards-doc/src/docs/antora/modules/isy-ueberwachung/pages/nutzungsvorgaben/inhalt.adoc

```diff
@@ -324,54 +326,40 @@ isy.task.tasks.isyHealthTask.fixed-delay=30s
 [[absicherung-von-endpoints]]
 ===== Absicherung von Endpoints
 
-Der Zugriff auf Endpoints muss mit einer Authentifizierung abgesichert werden.
-Eine Konfiguration für Spring Security, die alle Endpoints mit HTTP Basic Authentication absichert,
-zeigt <<listing-endpointsecurityconfig>>.
-Der Benutzername und das Passwort werden in `application.properties` gepflegt.
+Standardmäßig wird der Zugriff auf Endpoints durch eine HTTP Basic Authentifizierung abgesichert, wenn die Dependencies:
 
+ * `org.springframework.security:spring-security-config` und
+ * `org.springframework.security:spring-security-web`
 
-.Absicherung der Endpoints mit Spring Security
-[id="listing-endpointsecurityconfig",reftext="{listing-caption} {counter:listings }"]
-[source,java]
-----
-@Configuration
-@EnableWebSecurity
-@Profile("produktion")
-public class ActuatorSecurityProduktionConfig extends WebSecurityConfigurerAdapter {
+im Klassenpfad verfügbar sind.
 
-    @Autowired
-    private UeberwachungSecurityConfigProperties properties;
+Die Konfiguration von Benutzername und Passwort zum Zugriff auf die Endpoints ist optional und wird in der `application.properties` gepflegt (siehe <<listing-endpoint-security-properties>>).
+Ohne Konfiguration von Benutzername und Passwort bleibt die Absicherung aktiv und es kann nicht auf die Endpoints zugegriffen werden.
 
-    private static final String ENDPOINT_ROLE = "ENDPOINT_ADMIN";
+.Konfiguration der Absicherung in `application.properties`
+[id="listing-endpoint-security-properties",reftext="{listing-caption} {counter:listings }"]
+[source,properties]
+----
+# Benutzername des Monitoring Users (Default: Null)
+isy.ueberwachung.security.username=<username>
+# Passwort des Monitoring Users (Default: Null)
+isy.ueberwachung.security.password=<password>
+----
 
-    @Override
-    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
-         auth.inMemoryAuthentication()
-             .withUser(properties.getUsername())
-             .password(passwordEncoder().encode(properties.getPassword()))
-             .roles(ENDPOINT_ROLE);
-    }
+Die Standardabsicherung der Endpoints kann per Konfiguration deaktiviert werden:
 
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
-        http.requestMatcher(EndpointRequest.toAnyEndpoint())
-                .authorizeHttpRequests(requests -> requests.anyRequest().hasRole(ENDPOINT_ROLE))
-                .httpBasic();
-    }
+`spring.autoconfigure.exclude=de.bund.bva.isyfact.ueberwachung.autoconfigure.IsyActuatorSecurityAutoConfiguration`
 
-    @Bean
-    public PasswordEncoder passwordEncoder() {
-        return new BCryptPasswordEncoder();
-    }
-}
-----
+Dabei muss beachtet werden, dass über die Endpoints sensible Informationen bereitgestellt werden!
```

Comment:
Soll das bedeuten, dass mit einer Deaktivierung der Standardabsicherung die Endpoints ungesichert erreichbar sind?

Ich kann mir vorstellen, dass dies ggf. in Testumgebungen sinnvoll sein kann, doch für Produktiv-Umgebungen erscheint mir dieses Feature als zu gefährlich, wie es in Zeile 353 als beachtenswert beschrieben ist und auch in dem Ticket nicht als Anforderung besteht.

Falls dieses Feature erhalten bleiben soll, dann wäre m.E. eine Prüfung sinnvoll, die sicherstellt, dass in einer Produktiv-Umgebung die Absicherung immer aktiv sein muss.

Reply:
Das ist Teil von ISY-601 - bitte Abhängigkeit der Tickets beachten (siehe: https://github.com/IsyFact/isyfact-standards/pull/305#issuecomment-1991529451)

Die Abhängigkeit ist wie abgesprochen am Ticket und am PR kommentiert!

Ansonsten für die Anforderung siehe Lösungsbeschreibung zu A3 am Ticket ISY-601/IFS-3177